### PR TITLE
[SYCL] Fix FPGA devices selection on queue initialization

### PR DESCRIPTION
**Pull request by mlychkov**
FPGA device selectors use global constants of std::string type as
parameters in constructor. If some other inline global variable uses
fpga_device_selector in its constructor then there may be a situation
when FPGA device selector is initialized before global std::string
objects that it is depending on. Then it cannot find FPGA hw or
emulator devices because it has uninitialized platform name member.
Change type of constant strings with platform names to constant
C-string to guarantee initialization of FPGA device selector with
proper platform name.

### CODE REVIEW

File: sycl/include/CL/sycl/intel/fpga_device_selector.hpp [hunk 1]
```diff
@@ -32,9 +32,9 @@ class platform_selector : public device_selector {
   }
 };
 
-const std::string EMULATION_PLATFORM_NAME =
+const char *EMULATION_PLATFORM_NAME =
```
Comment (alexbatashev) [inline sycl/include/CL/sycl/intel/fpga_device_selector.hpp:35]: Is it more C++ now?
```suggestion
static constexpr auto EMULATION_PLATFORM_NAME =
```

Comment (v-klochkov) [reply sycl/include/CL/sycl/intel/fpga_device_selector.hpp:35]: 'auto' should be used only where the type is obvious, and this is not the case, imo.

File: sycl/test/fpga_tests/global_fpga_device_selector.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+//==--- global_fpga_device_selector.cpp - SYCL FPGA device selector test ---==//
+//
```
Comment (v-klochkov) [inline sycl/test/fpga_tests/global_fpga_device_selector.cpp:2]: the license section may be omitted for LIT tests.

File: sycl/include/CL/sycl/intel/fpga_device_selector.hpp (hunk 1 above)
Comment (MrSidims) [reply sycl/include/CL/sycl/intel/fpga_device_selector.hpp:35]: +1 to a comment above. If we really want it be C-plus-plusish - we can use std::array here (I personally don't have such preferences).

File: sycl/test/fpga_tests/global_fpga_device_selector.cpp (hunk 2 above)
Comment (mlychkov) [reply sycl/test/fpga_tests/global_fpga_device_selector.cpp]: Removed.

File: sycl/include/CL/sycl/intel/fpga_device_selector.hpp (hunk 1 above)
Comment (mlychkov) [reply sycl/include/CL/sycl/intel/fpga_device_selector.hpp]: I don't think it's worth including the std header just for a constant string that is used for one particular case.

File: sycl/include/CL/sycl/intel/fpga_device_selector.hpp [hunk 3]
```diff
@@ -32,9 +32,10 @@ class platform_selector : public device_selector {
   }
 };
 
-const std::string EMULATION_PLATFORM_NAME =
+static constexpr char const *EMULATION_PLATFORM_NAME =
     "Intel(R) FPGA Emulation Platform for OpenCL(TM)";
-const std::string HARDWARE_PLATFORM_NAME = "Intel(R) FPGA SDK for OpenCL(TM)";
+static constexpr char const *HARDWARE_PLATFORM_NAME =
```
Comment (v-klochkov) [inline sycl/include/CL/sycl/intel/fpga_device_selector.hpp:37]: 'constexpr + const' seems too much to me )) , but probably Ok.

Comment (keryell) [reply sycl/include/CL/sycl/intel/fpga_device_selector.hpp:37]: `static constexpr auto HARDWARE_PLATFORM_NAME =` is simpler

**Pull request merged**